liquibase-service/src/main/resources/db/rdb_modern/routines/023-sp_d_interview_postprocessing-001.sql

```diff
@@ -0,0 +1,614 @@
+CREATE OR ALTER PROCEDURE dbo.sp_d_interview_postprocessing(
+    @interview_uids NVARCHAR(MAX),
+    @debug bit = 'false')
+as
+
+BEGIN
+
+    DECLARE @RowCount_no INT;
+    DECLARE @Proc_Step_no FLOAT = 0;
+    DECLARE @Proc_Step_Name VARCHAR(200) = '';
+    DECLARE @ColumnAdd_sql NVARCHAR(MAX) = '';
+    DECLARE @PivotColumns NVARCHAR(MAX) = '';
+    DECLARE @Pivot_sql NVARCHAR(MAX) = '';
+    DECLARE @Insert_sql NVARCHAR(MAX) = '';
+    DECLARE @Update_sql NVARCHAR(MAX) = '';
+    -- number of columns for the dynamic sql
+    DECLARE @Col_number BIGINT = (SELECT COUNT(*) FROM dbo.nrt_metadata_columns);
+
+    BEGIN TRY
+
+        SET @Proc_Step_no = 1;
+        SET @Proc_Step_Name = 'SP_Start';
+        DECLARE @batch_id bigint;
+        SET @batch_id = cast((format(GETDATE(), 'yyMMddHHmmss')) AS bigint);
+
+        BEGIN TRANSACTION;
+
+        SELECT @ROWCOUNT_NO = 0;
+
+        INSERT INTO [DBO].[JOB_FLOW_LOG]
+        (BATCH_ID, [DATAFLOW_NAME], [PACKAGE_NAME], [STATUS_TYPE], [STEP_NUMBER], [STEP_NAME], [ROW_COUNT])
+        VALUES (@BATCH_ID, 'D_INTERVIEW', 'D_INTERVIEW', 'START', @PROC_STEP_NO, @PROC_STEP_NAME, @ROWCOUNT_NO);
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN TRANSACTION;
+
+        SET @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET @PROC_STEP_NAME = ' GENERATING #NEW_COLUMNS';
+
+        SELECT RDB_COLUMN_NM
+        INTO #NEW_COLUMNS
+        FROM dbo.nrt_metadata_columns
+        WHERE NEW_FLAG = 1
+          AND RDB_COLUMN_NM NOT IN (SELECT COLUMN_NAME
+                                    FROM INFORMATION_SCHEMA.COLUMNS
+                                    WHERE TABLE_NAME = 'D_INTERVIEW'
+                                      AND TABLE_SCHEMA = 'dbo');
+
+        SELECT @ROWCOUNT_NO = @@ROWCOUNT;
+
+        INSERT INTO [DBO].[JOB_FLOW_LOG]
+        (BATCH_ID, [DATAFLOW_NAME], [PACKAGE_NAME], [STATUS_TYPE], [STEP_NUMBER], [STEP_NAME], [ROW_COUNT])
+        VALUES (@BATCH_ID, 'D_INTERVIEW', 'D_INTERVIEW', 'START', @PROC_STEP_NO, @PROC_STEP_NAME, @ROWCOUNT_NO);
+
+        COMMIT TRANSACTION;
+
+
+        BEGIN TRANSACTION;
+
+        SET @PROC_STEP_NO = @PROC_STEP_NO + 1;
+        SET @PROC_STEP_NAME = 'ADDING COLUMNS TO D_INTERVIEW';
```

Comment:
👍

Reply:
Is this a step we might encounter often?

Reply:
Not a step we expect to see often. If there are no new columns, then it won't be performed.